src/Validator/Unique.php

```diff
@@ -25,60 +22,31 @@ class Unique extends Constraint
         self::NOT_UNIQUE_ERROR => 'NOT_UNIQUE_ERROR',
     ];
 
-    public $message = 'This value is already used.';
+    public string $message = 'This value is already used.';
 
     /**
-     * @var string
+     * @param non-empty-array<int|string, string> $fields      DTO -> entity field mapping
+     * @param array<int|string, string>           $idFields    DTO -> entity ID field mapping
+     * @param class-string                        $entityClass
      */
-    public $entityClass;
-
-    /**
-     * DTO -> entity field mapping.
-     *
-     * @var string[]
-     */
-    public $fields;
-
-    /**
-     * DTO -> entity ID field mapping.
-     *
-     * @var string[]|null
-     */
-    public $idFields;
-
-    public $errorPath = '';
-
-    public function __construct($options = null)
-    {
-        parent::__construct($options);
-
-        $fields = $options['fields'] ?? $options['value'];
-
-        if (!\is_array($fields) && !\is_string($fields)) {
-            throw new UnexpectedTypeException($fields, 'array or string');
-        }
-
-        $fields = (array) $fields;
+    #[HasNamedArguments]
+    public function __construct(
+        public string $entityClass,
+        public string $errorPath,
+        public array $fields,
+        public array $idFields = [],
+    ) {
+        parent::__construct([]);
 
         if (0 === \count($fields)) {
-            throw new InvalidOptionsException('fields option must have at least one field', ['fields']);
+            throw new InvalidOptionsException('`fields` option must have at least one field', ['fields']);
         }
 
-        if (!$options['entityClass']) {
+        if ('' === $this->entityClass) {
```

Comment:
I'm wondering if this also needs to do `null === $this->entityClass || '' === $this->entityClass)` or that gets checked elsewhere. I'm at least still able to hit unique validator errors so perhaps I'm mistaken, just noticed other things did when doing `'' ===`

Reply:
I agree